nuleaf-source/database/posts.js

```diff
@@ -0,0 +1,161 @@
+ *                   created_before  : Filter for posts after this date.
+ *                   created_after   : Filter for posts before this date.
+ *                   published_before: Filter for posts after this date.
+ *                   published_after : Filter for posts before this date.
+ *                   modified_before : Filter for posts after this date.
+ *                   modified_after  : Filter for posts before this date.
+ *
+ * @return {Error}, {Integer} Integer of posts with matching conditions.
+ */
+ exports.count = function(conditions, callback) {
+  if (typeof conditions === 'function') {
+    callback = conditions;
+  conditions = {};
+  }
+
+  var _conditions = buildConditions(conditions);
+  Post.count(_conditions, callback);
+};
+
+/**
+ * Creates a new post and returns it.
+ * @param {Object} Post data.
+ * 
+ * @return {Error}, {Post} Post if created or null.
+ */
+exports.create = function(data, callback) {
+  Post.create(data, callback);
+};
+
+/**
+ * Find a post with unique id.
+ * @param {ObjectId} Unique id of the post.
+ *
+ * @return {Error}, {Post} Post if found or null.
+ */
+exports.get = function(id, callback) {
+  Post.findById(id, callback);
+};
+
+/**
+ * Find a post with unique id and updates it.
+ * @param {Object} Post id along with updates.
+ *
+ * @return {Error}, {Post} The updated post
+ */
+exports.update = function(update, callback) {
+  update = Helper.filterOutEmpty(update);
+  var id = update.id;
+  delete update.id;
+
+  Post.findByIdAndUpdate(id, update, callback);
+};
+
+/**
+ * Deletes a post from the collection via its id.
+ * @param {ObjectsId} Post id to delete.
+ *
+ * @return {Error}
+ */
+exports.delete = function(id, callback) {
+  Post.remove({ _id: id }, callback);
+};
+
+/**
+ * Filters and set conditions.
+ * @param {Object} Conditions
+ *
+ * @return {Object} Conditions.
+ */
+function buildConditions(conditions) {
+  var _conditions = {};
+
+  // Search for these fields using Regular Expression search.
+  if (conditions.title) { _conditions.title = new RegExp(conditions.title, 'i'); }
+  if (conditions.content) { _conditions.content = new RegExp(conditions.content, 'i'); }
+  if (conditions.author) { _conditions.author = new RegExp(conditions.author, 'i'); }
+  
+  if (conditions.created_before || conditions.created_after) {
+    _conditions.date_created = getDateRangeQuery(conditions.created_before, conditions.created_after);
+  }
+  if (conditions.modified_before || condtions.modified_after) {
+    _conditions.date_modified = getDateRangeQuery(conditions.modified_before, conditions.modified_after);
+  }
+  if (conditions.published_before || conditions.published_after) {
+    _conditions.date_published = getDateRangeQuery(conditions.published_before, conditions.published_after);
+  }
+
+  return _conditions;
+}
+
+/**
+ * Searches for the date range.
+ * @param {Object} date before and date after
```

Comment:
Not that important but there are two parameters here and they are both a `Date` object.